src/fmeval/eval_algorithms/common.py

```diff
@@ -101,31 +101,32 @@ def evaluate_dataset(
 
     :return: An EvalOutput object encapsulating the results of the evaluation.
     """
-    if model:
-        try:
-            validate_dataset(dataset, [DatasetColumns.MODEL_INPUT.value.name])
-        except EvalAlgorithmClientError:
-            raise EvalAlgorithmClientError(
-                "evaluate_dataset has been given a ModelRunner to obtain outputs from "
-                "but the provided dataset does not contain a model input column."
-            )
-        prompt_template = get_default_prompt_template(dataset_name) if not prompt_template else prompt_template
-        model_invocation_pipeline = create_model_invocation_pipeline(model, prompt_template)
-        pipeline = TransformPipeline([model_invocation_pipeline, pipeline])
-    else:
-        if prompt_template:
-            logger.warning(
-                "A prompt template, but no corresponding model, was provided."
-                "Model outputs from the dataset will be used, and this prompt template will be ignored."
-            )
-        try:
-            validate_dataset(dataset, [DatasetColumns.MODEL_OUTPUT.value.name])
-        except EvalAlgorithmClientError:
-            raise EvalAlgorithmClientError(
-                "evaluate_dataset has been given a dataset with no model output column "
-                "and no ModelRunner to obtain outputs from. Please either provide a model "
-                "or use a dataset that contains model outputs already."
-            )
+    if eval_name != EvalAlgorithm.CONTEXT_QUALITY.value:
```

Comment:
Why do we need this `if`? Assuming that it is needed, is there somewhere else in the code that we can move it? It feels a bit inelegant to handle this in `evaluate_dataset`, which is meant to be very generic (agnostic to which eval algorithm is being run).

Reply:
I don't think this check is necessary, as in context_quality eval algo, we won't pass `model` and `prompt_template` to `evaluate_dataset` method, so this block of code won't be executed anyway

Reply:
The check is currently necessary, as the `else` case from the `if model` will trigger, and then `validate_dataset(dataset, [DatasetColumns.MODEL_OUTPUT.value.name])` will fail.

Reply:
We could add a boolean parameter `validate_columns` to `evaluate_dataset` to specify if the validations need to be run for a eval algo.